torch_xla/distributed/xla_backend.py

```diff
@@ -82,10 +82,8 @@ def allgather(self, output_tensors_list, input_tensors):
   # https://github.com/pytorch/pytorch/blob/release/1.10/torch/distributed/distributed_c10d.py#L1129
   def broadcast(self, tensors, opts):
     root_tensor = tensors[opts.rootTensor]
-    root_rank = opts.rootRank
-    if root_rank != self.rank():
-      with torch.no_grad():
-        root_tensor.zero_()
+    scale = 1 if opts.rootRank != self.rank() else 0
+    root_tensor = root_tensor * scale
     xm.all_reduce(
```

Comment:
Hmm, does XLA have broadcast? Maybe a proper fix is to introduce the XLA broadcast instead of faking it using all_reduce?

Reply:
XLA does.. but XLA broadcast is not cross replica, it is broadcast in a different context. check https://www.tensorflow.org/xla/operation_semantics#broadcast

Reply:
Is this the collective broadcast? To me. it seems to be the other broadcast?

Reply:
it is not the collective one, XLA does not have a collective broadcast AFAIK.